docs/current/guides/757394-use-service-containers.md

```diff
@@ -54,167 +54,290 @@ Service containers come with the following built-in features:
 
 </TabItem>
-</Tabs>
+<TabItem value="Node.js">
 
-## Bind services
+TODO: This example calls `Host.service([]PortForward)` to create a service that proxies traffic through the host to the configured ports. It then sets the service binding on the client container to the host.
+
+</TabItem>
+<TabItem value="Python">
 
-Dagger enables users to bind a service container to a client container with an alias (such as `redis`) that the client container can use as a hostname.
+TODO: This example calls `Host.service([]PortForward)` to create a service that proxies traffic through the host to the configured ports. It then sets the service binding on the client container to the host.
 
-Binding a service to a container expresses a dependency: the service container needs to be running when the client container runs. The bound service container is started automatically whenever its client container runs.
+</TabItem>
+</Tabs>
 
-Here's an example of an HTTP service automatically starting in tandem with a client container. The service binding enables the client container to access the HTTP service using the alias `www`.
+:::note
+The Dagger implementation is a new flavor of `Socket` and is similar to `Host.unixSocket`.
+:::
 
-<Tabs groupId="language" className="embeds">
+## Persist service state
+
+Another way to avoid relying on the grace period is to use a cache volume to persist a service's data, as in the following example:
+
+<Tabs groupId="language">
 <TabItem value="Go">
 
-<Embed id="pQoE-5_0Ghg" />
+```go file=./snippets/use-services/persist-service-state/main.go
+```
 
 </TabItem>
 <TabItem value="Node.js">
 
-<Embed id="VIFYGYc8YRN" />
+```typescript file=./snippets/use-services/persist-service-state/index.ts
+```
 
 </TabItem>
 <TabItem value="Python">
 
-<Embed id="FkPqDoW3Oo-" />
+```python file=./snippets/use-services/persist-service-state/main.py
+```
 
 </TabItem>
 </Tabs>
 
-:::tip
-Services in service containers should be configured to listen on the IP address 0.0.0.0 instead of 127.0.0.1. This is because 127.0.0.1 is only reachable within the container itself, so other services (including the Dagger health check) won't be able to connect to it. Using 0.0.0.0 allows connections to and from any IP address, including the container's private IP address in the Dagger network.
+:::info
+This example uses Redis's `SAVE` command to ensure data is synced. By default, Redis flushes data to disk periodically.
 :::
 
-When a service is bound to a container, it also conveys to any outputs of that container, such as files or directories. The service will be started whenever the output is used, so you can also do things like this:
+## Start and stop services
+
+Starting with Dagger v0.8.8, you can explicitly start and stop services in your pipelines.
 
-<Tabs groupId="language" className="embeds">
+<Tabs groupId="language">
 <TabItem value="Go">
 
-<Embed id="Qoljk5SEPuu" />
+TODO
```

Comment:
@vito can you please provide an example? In any one language is fine, I will convert for the other SDKs.